### DISCUSSION TITLE
Including "unpaid interest" when margin<1?

### DISCUSSION THREAD

**Discussion by zaferyumruk** [kernc/backtesting.py#590] [Ideas]
Greetings,

How can "unpaid interest" be added to the backtest?

**Option 1**
Defining `Position.amount_borrowed` and `Position.unpaid_interest`? 

`interest rate` will be contained in market data. While calculating `_Broker.equity` `Position.unpaid_interest` will be included along with pl. I am puzzled about paying `unpaid_interest` though. Since the library focused on managing trades, paying interests should be maintained through closing or reducing active trades which is not possible for this option.

**Option 2**
Passing `borrowed_amount` for each trade when order executed. 

Then on each `_Broker._close_trade` or `_Broker._reduce_trade` respective interest will be paid by reducing `_Broker.cash`. For exclusive_orders and hedging cases I am not sure "excess borrowing" (i.e while closing a trade that borrowed zero amount, keeping loans from other active trades; keeping debt while you have cash thus paying unnecessary interest) occurs.

Thanks in advance.

**Comment by kernc** [comment]
> Defining `Position.amount_borrowed` and `Position.unpaid_interest`?

Isn't this something that could be easily enough computed from [trades data](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Trade-data) (namely `Size`, `EntryPrice` and `Duration`, and `margin=`) after the backtest?

**Reply by zaferyumruk** [reply]
Thank you for the response!

Sure it is. But what about reflecting the effect of intermediate interest payments on the whole backtest? These payments will reduce intermediate equities. And this will affect the whole trade cumulatively. Do you think a cumulative down-scaling could work? (i.e finding ratio `1‑intermediate_interest_paid/equity` for each entry on the equity curve and reducing the final return using the products of these ratios)

Reflecting these to `broker.closed_trades` is another challenge if computing statistics (`compute_stats`) is also essential. A similar challenge can also be mentioned for accurate `DrawdownPct` in `equity` data.

Am I just overthinking?

**Reply by zaferyumruk** [reply]
margin used can be extracted as you suggested. Thus holding cost can be subtracted from equity with each repayment. 

With each repayment, the rest of the equity can be downscaled to reflect the reduction of the invested amount. 

Thanks for the suggestion,

Great library!
